src/main/java/com/google/sps/servlets/ImageHandlerServlet.java

```diff
@@ -76,15 +90,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
   }
 
-  private Entity newPlayerEntity(String imageBlobKeyString, String displayName) {
-    user = UserServiceFactory.getUserService().getCurrentUser();
-    player = new Entity(PLAYER_QUERY_PARAMETER);
-    player.setProperty(EMAIL_PARAMETER, user.getEmail());
-    player.setProperty(IMAGE_ID_PARAMETER, imageBlobKeyString);
-    player.setProperty(DISPLAY_NAME_PARAMETER, displayName);
-    player.setProperty(ID_PARAMETER, user.getUserId());
-    player.setProperty(EXPERIENCE_POINTS_PARAMETER, NO_EXPERIENCE);
-    player.setProperty(PROMOTION_THRESHOLD_PARAMETER, STARTER_THRESHOLD);
+  private Player newPlayer(String imageBlobKeyString, String displayName) {
+    User user = UserServiceFactory.getUserService().getCurrentUser();
+    Player player =
+        new Player(displayName, user.getEmail(), user.getUserId(), imageBlobKeyString, START_PAGE);
+    player.setExperiencePoints(NO_EXPERIENCE);
+    player.setPromotionThreshold(STARTER_THRESHOLD);
```

Comment:
Was it the entities that where creating all those bugs and that is the reason we switched to Player objects ?

Reply:
I thought it would be better and safer to create a Player object then have the PlayerDatabase handle converting it to an entity. More abstraction/ safeguard checks that way.